app/component/SummaryPage/hoc/withFetchMoreItineraries.js

```diff
@@ -0,0 +1,312 @@
+
+/**
+ * Clamp value under zero to zero.
+ *
+ * @param {Number} value
+ * @returns {boolean}
+ */
+export const clamp = value => Math.max(0, value);
+
+const useAutofetchEarlier = (relayEnvironment, onData, onComplete) => {
+  const { fetch } = useAutofetchQuery({
+    relayEnvironment,
+    query: moreItinerariesQuery,
+    initialState: {
+      missingCount: NUMBER_MORE_ITINERARIES,
+    },
+    getPageCursor: data => data.plan.previousPageCursor,
+    onData,
+    onComplete,
+    updateState: (data, state) => ({
+      ...state,
+      missingCount: clamp(state.missingCount - data.plan.itineraries.length),
+    }),
+    updateVariables: (variables, state) => ({
+      ...variables,
+      numItineraries: state.missingCount,
+    }),
+    shouldFetchMore: state => state.missingCount > 0,
+  });
+  return fetch;
+};
+
+const useAutofetchLater = (relayEnvironment, onData, onComplete) => {
+  const { fetch } = useAutofetchQuery({
+    relayEnvironment,
+    query: moreItinerariesQuery,
+    initialState: {
+      missingCount: NUMBER_MORE_ITINERARIES,
+    },
+    getPageCursor: data => data.plan.nextPageCursor,
+    onData,
+    onComplete,
+    updateState: (data, state) => ({
+      ...state,
+      missingCount: clamp(state.missingCount - data.plan.itineraries.length),
+    }),
+    updateVariables: (variables, state) => ({
+      ...variables,
+      numItineraries: state.missingCount,
+    }),
+    shouldFetchMore: state => state.missingCount > 0,
+  });
+  return fetch;
+};
+
+const withItineraryPaging = WrappedComponent => {
+  const HOCWrapperComponent = (props, context) => {
+    const isParkAndRideSelected = props.match.params.hash === 'parkAndRide';
+
+    const [loadingMoreItineraries, setLoadingMoreItineraries] = useState();
+    const [earlierItineraries, setEarlierItineraries] = useState([]);
+    const [laterItineraries, setLaterItineraries] = useState([]);
+    const [separatorPosition, setSeparatorPosition] = useState();
+    const [error, setError] = useState();
+
+    const resetSummaryPageSelection = () => {
+      context.router.replace({
+        ...context.match.location,
+        state: {
+          ...context.match.location.state,
+          summaryPageSelected: undefined,
+        },
+      });
+    };
+
+    const showScreenreaderLoadedAlert = () => {
+      props.srPushAlert(
+        'itinerary-page.itineraries-loaded',
+        'More itineraries loaded',
+      );
+    };
+
+    const onDataEarlier = (data, state) => {
+      const {
+        plan: { itineraries },
+      } = data;
+
+      if (state.arriveBy) {
+        setLaterItineraries([...laterItineraries, ...itineraries]);
+      } else {
+        // Reverse the results so that route suggestions are in ascending order
+        const reversedItineraries = itineraries
+          .slice() // Need to copy because result is readonly
+          .reverse()
+          .filter(
+            itinerary => !itinerary.legs.every(leg => leg.mode === 'WALK'),
```

Comment:
I think we could actually leave the WALK mode out of the public transportation requests because OTP2 returns walk legs for public transportation even without it. Then we wouldn't need to do this filtering in UI which could simplify the logic a bit.

Reply:
We won't do this because the WALK only itineraries are now sometimes rendered.